src/System.Runtime.Extensions/tests/System/Diagnostics/Stopwatch.cs

```diff
@@ -93,6 +94,35 @@ public static void StartNewAndRestart()
             }
         }
 
+        [OuterLoop("Sleeps for relatively long periods of time")]
+        [Fact]
+        public static void ElapsedMilliseconds_WithinExpectedWindow()
+        {
+            const int AllowedTries = 30;
+            const int SleepTime = 1000;
+            const double WindowFactor = 2;
+
+            var results = new List<long>();
+
+            var sw = new Stopwatch();
+            for (int trial = 0; trial < AllowedTries; trial++)
+            {
+                sw.Restart();
+                Thread.Sleep(SleepTime);
+                sw.Stop();
+
+                if (sw.ElapsedMilliseconds >= (SleepTime / WindowFactor) &&
+                    sw.ElapsedMilliseconds <= (SleepTime * WindowFactor))
+                {
+                    return;
```

Comment:
Isn't this itself potentially flaky, if the first loop happens to fall in the range? In that scenario, are you expect subsequent CI to then hit it?

Reply:
It's a stop-gap.  It could have false negatives, but if there's actually a problem where we're off by a factor of 10, then some run of the test should flag it.